src/python/zquantum/core/openfermion/_qiskit_conversions.py

```diff
@@ -0,0 +1,88 @@
+############################################################################
+#   Copyright 2017 Rigetti Computing, Inc.
+#   Modified by Zapata Computing 2020 to work for qiskit's WeightedPauliOperator.
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+############################################################################
+
+"""
+Translates OpenFermion Objects to qiskit WeightedPauliOperator objects
+"""
+from openfermion import QubitOperator, count_qubits
+from typing import Union
+from qiskit.aqua.operators import WeightedPauliOperator
+from qiskit.quantum_info import Pauli
+
+
+def qubitop_to_qiskitpauli(qubit_operator: QubitOperator) -> WeightedPauliOperator:
+    """
+    Convert a OpenFermion QubitOperator to a WeightedPauliOperator.
+
+    Args:
+        qubit_operator: OpenFermion QubitOperator to convert to a qiskit.aqua.operators.WeightedPauliOperator
+
+    Returns:
+        WeightedPauliOperator representing the qubit operator
+    """
+    if not isinstance(qubit_operator, QubitOperator):
+        raise TypeError("qubit_operator must be a OpenFermion " "QubitOperator object")
```

Comment:
I think there are some legacy reasons why we were doing type checking. Should we remove this and rely on type hinting instead?

Reply:
I think one reason why we're doing it might be the fact that if you try to do it on `IsingOperator` it should work in principle but does not in practice and it's confusing as hell, so to make it more explicit we're throwing an error?
Or you think that was sth else?
Though it seems it should be working with `IsingOperator` as well 🤔

Reply:
I know there were cases historically where we had to distinguish `QubitOperator` from `IsingOperator`. But in this case it looks like the type checking came from from [forest-openfermion](https://github.com/rigetti/forest-openfermion/blob/53d0a9c7ebac1e540c793e3217593298d422aec2/forestopenfermion/pyquil_connector.py#L31-L33).

Reply:
Actually, another good reason to have this type check is because `count_qubits` doesn't work for `IsingOperator`. 
I did a PR, see here: https://github.com/quantumlib/OpenFermion/pull/712 .